### PR TITLE
Android: Add descriptors, make HowToFix a PropertyBag

### PR DESCRIPTION
**Pull request by DaveTryon**
#### Description of changes

(give an overview. Add technical description describing your changes)

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/unit/tests/electron/platform/android/__snapshots__/scan-results-to-unified-results.test.ts.snap [hunk 1]
```diff
@@ -4,43 +4,86 @@ exports[`ScanResultsToUnifiedResults Null ScanResults input returns empty output
...
+      "boundingRectangle": Object {
+        "bottom": 800,
+        "left": 10,
+        "right": 600,
+        "top": 0,
+      },
+      "className": "myClass1",
+      "contentDescription": "myDescription1",
+      "text": "myText1",
+    },
```
Comment (DaveTryon) [inline src/tests/unit/tests/electron/platform/android/__snapshots__/scan-results-to-unified-results.test.ts.snap]: @karanbirsingh, please let me know if there are other properties you think we ought to have in the descriptor properties

File: src/tests/unit/tests/electron/platform/android/__snapshots__/scan-results-to-unified-results.test.ts.snap [hunk 2]
```diff
@@ -4,43 +4,86 @@ exports[`ScanResultsToUnifiedResults Null ScanResults input returns empty output
...
     "identifiers": null,
     "resolution": Object {
-      "how-to-fix": "The text element has insufficient contrast of 1. Foreground color: #ffffff, background color: #ffffff). Modify the text foreground and/or background colors to provide a contrast ratio of at least 4.5:1 for regular text, or 3:1 for large text (at least 18pt, or 14pt+bold).",
+      "how-to-fix": Object {
+        "unformattedText": "The text element has insufficient contrast of 1. Foreground color: #ffffff, background color: #ffffff). Modify the text foreground and/or background colors to provide a contrast ratio of at least 4.5:1 for regular text, or 3:1 for large text (at least 18pt, or 14pt+bold).",
+      },
     },
     "ruleId": "ColorContrast",
```
Comment (DaveTryon) [inline src/tests/unit/tests/electron/platform/android/__snapshots__/scan-results-to-unified-results.test.ts.snap]: @karanbirsingh, this is just the unformatted text for the HowToFix. We can later add a formatted experience at a future time, as needed. Please let me know if you think we need to include more that this point.

File: src/tests/unit/tests/electron/platform/android/__snapshots__/scan-results-to-unified-results.test.ts.snap [hunk 3]
```diff
@@ -4,43 +4,86 @@ exports[`ScanResultsToUnifiedResults Null ScanResults input returns empty output
...
     "status": "fail",
     "uid": "gguid-mock-stub",
   },
   Object {
-    "descriptors": null,
+    "descriptors": Object {
+      "boundingRectangle": null,
+      "className": "myClass3",
+      "contentDescription": null,
+      "text": null,
```
Comment (DaveTryon) [inline src/tests/unit/tests/electron/platform/android/__snapshots__/scan-results-to-unified-results.test.ts.snap]: What's the consensus on returning null values in properites? Is it preferable to simply omit the keys if the values are null?

File: src/tests/unit/tests/electron/platform/android/__snapshots__/scan-results-to-unified-results.test.ts.snap (hunk 1 above)
Comment (karanbirsingh) [reply src/tests/unit/tests/electron/platform/android/__snapshots__/scan-results-to-unified-results.test.ts.snap]: I think these are good

File: src/tests/unit/tests/electron/platform/android/__snapshots__/scan-results-to-unified-results.test.ts.snap (hunk 3 above)
Comment (smoralesd) [reply src/tests/unit/tests/electron/platform/android/__snapshots__/scan-results-to-unified-results.test.ts.snap]: I think this is fine... the class using `descriptors` needs to handle the `text` being `null`

File: src/tests/unit/tests/electron/platform/android/rule-information-provider.test.ts [hunk 4]
```diff
@@ -17,44 +18,44 @@ describe('RuleInformationProvider', () => {
...
+    function validateHowToFix(ruleId: string, ruleResult: RuleResultsData): InstancePropertyBag {
         const ruleInformation: RuleInformation = provider.getRuleInformation(ruleId);
-        const howToFix: string = ruleInformation.howToFix(ruleResult);
+        const howToFix = ruleInformation.howToFix(ruleResult);
 
         expect(ruleInformation).toBeTruthy();
         expect(ruleInformation.ruleId).toEqual(ruleId);
         expect(ruleInformation.ruleDescription.length).toBeGreaterThan(0);
-        expect(howToFix.length).toBeGreaterThan(0);
+        expect(howToFix).toBeTruthy();
```
Comment (smoralesd) [inline src/tests/unit/tests/electron/platform/android/rule-information-provider.test.ts]: this `expect`  seems "duplicated" with the `expect(howToFix).toMatchSnapshot();` on the test themselves

File: src/tests/unit/tests/electron/platform/android/rule-information.test.ts [hunk 5]
```diff
@@ -21,9 +24,20 @@ describe('RuleInformation', () => {
...
+            ruleId: 'some rule',
+            status: 'pass',
+            props: null,
+        };
+
         for (const howToFixString of testInputs) {
-            const ruleInformation = new RuleInformation(null, null, () => howToFixString);
-            expect(howToFixString === ruleInformation.howToFix(null));
+            const expectedPropertyBag = { someLabel: howToFixString };
+            const mock = Mock.ofType<HowToFixDelegate>();
```
Comment (smoralesd) [inline src/tests/unit/tests/electron/platform/android/rule-information.test.ts]: should we rename `mock` to something like `howToFixDelegateMock`?

File: src/tests/unit/tests/electron/platform/android/rule-information-provider.test.ts (hunk 4 above)
Comment (DaveTryon) [reply src/tests/unit/tests/electron/platform/android/rule-information-provider.test.ts]: Removed

File: src/tests/unit/tests/electron/platform/android/rule-information.test.ts (hunk 5 above)
Comment (DaveTryon) [reply src/tests/unit/tests/electron/platform/android/rule-information.test.ts]: Renamed as suggested

**Pull request merged**